src/containerapp/azext_containerapp/custom.py

```diff
@@ -2638,3 +2642,87 @@ def list_environment_telemetry_otlp(cmd,
 
     return containerapp_env_def
 
+
+def list_dotnet_components(cmd, environment_name, resource_group_name):
+    raw_parameters = locals()
+    dotnet_component_decorator = DotNetComponentDecorator(
+        cmd=cmd,
+        client=DotNetComponentPreviewClient,
+        raw_parameters=raw_parameters,
+        models=CONTAINER_APPS_SDK_MODELS
+    )
+    return dotnet_component_decorator.list()
+
+
+def show_dotnet_component(cmd, dotnet_component_name, environment_name, resource_group_name):
+    raw_parameters = locals()
+    dotnet_component_decorator = DotNetComponentDecorator(
+        cmd=cmd,
+        client=DotNetComponentPreviewClient,
+        raw_parameters=raw_parameters,
+        models=CONTAINER_APPS_SDK_MODELS
+    )
+    result = dotnet_component_decorator.show()
+    if result is not None:
+        logger.warning("Found DotNet Component '%s' in environment '%s' in resource group '%s'.", dotnet_component_name, environment_name, resource_group_name)
+
+    aspire_dashboard_url = _get_aspire_dashboard_url(cmd, environment_name, resource_group_name, dotnet_component_name)
+    component_type = safe_get(result, "properties", "componentType")
+    if component_type == DOTNET_COMPONENT_RESOURCE_TYPE:
+        logger.warning("Aspire Dashboard URL: %s.", aspire_dashboard_url)
```

Comment:
The logic about gets `aspire_dashboard_url` should be moved tunder `if component_type == DOTNET_COMPONENT_RESOURCE_TYPE:`, right?
```
    aspire_dashboard_url = _get_aspire_dashboard_url(cmd, environment_name, resource_group_name, dotnet_component_name)
    component_type = safe_get(result, "properties", "componentType")
```

Reply:
Updated.